art/estimators/classification/classifier.py

```diff
@@ -174,10 +215,38 @@ def save(self, filename: str, path: Optional[str] = None) -> None:
         """
         raise NotImplementedError
 
+    def clone_for_refitting(self) -> "ClassifierNeuralNetwork":  # lgtm [py/inheritance/incorrect-overridden-signature]
+        """
+        Create a copy of the estimator that can be refit from scratch.
+
+        :return: new estimator
+        """
+        raise NotImplementedError
+
+    def reset(self) -> None:
+        """
+        Resets the weights of the estimator so that it can be refit from scratch.
+
+        """
+        raise NotImplementedError
+
 
 class ClassifierDecisionTree(DecisionTreeMixin, ClassifierMixin, BaseEstimator, ABC):
     """
     Typing variable definition.
     """
 
-    pass
+    def clone_for_refitting(self) -> "ClassifierDecisionTree":  # lgtm [py/inheritance/incorrect-overridden-signature]
+        """
+        Create a copy of the estimator that can be refit from scratch.
+
+        :return: new estimator
+        """
+        raise NotImplementedError
+
+    def reset(self) -> None:
+        """
+        Resets the weights of the estimator so that it can be refit from scratch.
+
+        """
+        raise NotImplementedError
```

Comment:
I think these lines should not be necessary.

Reply:
Ok, I removed them from ClassifierLossGradients, ClassifierClassLossGradients, ClassifierNeuralNetwork and ClassifierDecisionTree. Should I leave them in Classifier or remove as well?

Reply:
@abigailgold Yes, because they are defined as abstract methods in `BaseEstimator` all estimators will already have inherited then.

Reply:
Just to make sure, you're saying I should remove them from the Classifier class as well?